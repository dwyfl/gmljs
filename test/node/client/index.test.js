import GML from '../../../src';
import GMLClient from '../../../src/node/client';
import testXml from './data/test.xml';

it('creates a correct default GMLClient node', () => {
  const _now = Date.now;
  Date.now = jest.fn(() => 0);
  try {
    const gml = GMLClient.create().toString();
    expect(gml).toMatchSnapshot();
  } finally {
    Date.now = _now;
  }
});

it('creates a correct GMLClient node from spec XML', () => {
  const gmlNode = GML.createNodeFromXml(GMLClient, testXml);
  const gml = gmlNode.toString();
  expect(gml).toMatchSnapshot();
});