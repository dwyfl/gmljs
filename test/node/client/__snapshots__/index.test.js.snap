// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a correct GMLClient node from spec XML 1`] = `"<client><name>Laser Tag</name><version>2.0</version><username>MyUserName</username><permalink>http://000000book.com/data/156/</permalink><keywords>katsu,paris,2010</keywords><uniquekey>28sks922ks992</uniquekey><ip>192.168.1.1</ip><time>1928372722</time><location><lon>-39.392922</lon><lat>53.29292</lat></location></client>"`;

exports[`creates a correct default GMLClient node 1`] = `"<client><name>GML.js</name><version>1.0</version><time>0</time></client>"`;
